packages/DateRangePicker/src/DateRangePicker.js

```diff
@@ -0,0 +1,238 @@
+    if (event.key === "Escape") {
+      hidePopover();
+    }
+  }
+
+  function onDateInputFocus(type) {
+    return () => {
+      setCurrentFocus(type);
+    };
+  }
+
+  function onFocusChange(type) {
+    if (type === START_DATE) {
+      startDateInputRef.current.focus();
+    } else if (type === END_DATE) {
+      endDateInputRef.current.focus();
+    } else {
+      hidePopover();
+    }
+  }
+
+  function onChangeStartDate(newDate) {
+    onDatesChange({ startDate: newDate, endDate });
+  }
+
+  function onChangeEndDate(newDate) {
+    onDatesChange({ startDate, endDate: newDate });
+  }
+
+  function shouldDenyStartDateInputConfirmation() {
+    return isElementContainsFocus(startDateInputRef.current);
+  }
+
+  function shouldDenyEndDateInputConfirmation() {
+    return isElementContainsFocus(endDateInputRef.current);
+  }
+
+  function handlePossibleStartDateChange(newPossibleDate) {
+    if (newPossibleDate.isSame(possibleStartDate, "year") && newPossibleDate.isSame(possibleStartDate, "month")) return;
+
+    setPossibleStartDate(newPossibleDate);
+  }
+
+  function handlePossibleEndDateChange(newPossibleDate) {
+    if (newPossibleDate.isSame(possibleEndDate, "year") && newPossibleDate.isSame(possibleEndDate, "month")) return;
+
+    setPossibleEndDate(newPossibleDate);
+  }
+
+  return (
+    <Popover
+      edge={AlignTypes.LEFT}
+      maxWidth="none"
+      offset={8}
+      {...popoverProps}
+      isOpen={shouldShowPopover}
+      onKeyUp={handleKeyUpOnEscape}
+      onClose={handleClosePopover}
+      shouldKeepFocus
+    >
+      <sc.Wrapper>
+        <sc.DateInput
+          ref={startDateInputRef}
+          date={startDate}
+          onChange={onChangeStartDate}
+          onChangePossibleDate={handlePossibleStartDateChange}
+          placeholder={startInputPlaceholder}
+          onClick={showPopover}
+          beforeConfirmation={handleClosePopover}
+          denyConfirmation={shouldDenyStartDateInputConfirmation}
+          onFocus={onDateInputFocus(START_DATE)}
+          {...inputProps}
+          {...startInputProps}
+        />
+        <sc.Separator>
+          <ArrowIcon />
+        </sc.Separator>
+        <sc.DateInput
+          ref={endDateInputRef}
+          date={endDate}
+          onChange={onChangeEndDate}
+          onChangePossibleDate={handlePossibleEndDateChange}
+          placeholder={endInputPlaceholder}
+          onClick={showPopover}
+          beforeConfirmation={handleClosePopover}
+          denyConfirmation={shouldDenyEndDateInputConfirmation}
+          onFocus={onDateInputFocus(END_DATE)}
+          {...inputProps}
+          {...endInputProps}
+        />
+      </sc.Wrapper>
+
+      <Popover.Content ref={popoverContentRef}>
+        <Popover.Card>
+          {shouldShowPopover && (
+            <sc.PopoverCardContent>
+              <sc.CalendarsWrapper>
+                <sc.CalendarWrapper data-pka-anchor="daterangepicker.leftcalendar">
+                  <Calendar
```

Comment:
I think was a good idea to split the `<Calendar />` to its own component reduced a lot of complexity while reading the code.